rasterio/_base.pyx

```diff
@@ -612,6 +611,38 @@ cdef class DatasetReader(object):
             factors.append(int(round(float(self.width)/float(xsize))))
         return factors
 
+    def checksum(self, bidx, window=None):
+        """Compute an integer checksum for the stored band
+
+        Parameters
+        ----------
+        bidx : int
+            The band's index (1-indexed).
+        window: tuple, optional
+            A window of the band. Default is the entire extent of the band.
+
+        Returns
+        -------
+        An int.
+        """
+        cdef void *hband = NULL
+        cdef xoff, yoff, width, height
```

Comment:
should this be ```cdef int xoff, yoff, width, height```?  Doesn't throw an error or warning during Cythonizing, but seems like a good idea to be explicit about type here.